justfile

```
@@ -0,0 +1,364 @@
+    @echo "  docker-up        - Start all services with docker-compose"
+    @echo "  docker-down      - Stop all services"
+
+    @echo "\nDeveloper Focused:"
+    @echo "  deps             - Install required dependencies"
+    @echo "  gen-proto        - Generate protobuf code"
+    @echo "  gen-templ        - Generate go code from templ files"
+    @echo "  gen-test-keys    - Generate test keys for MINT_PRIVATE_KEY"
+    @echo "  dev              - Start database and run application locally"
+    @echo "  test             - Run tests"
+    @echo "  lint             - Run linter"
+    @echo "  clean-all        - Clean build artifacts and cache"
+    @echo "  docker-db        - Start only the database service"
+    @echo "  docker-db-down   - Stop the database service"
+    @echo "  docker-clean     - Clean up Docker resources"
+
+    @echo "\nBuild and Release:"
+    @echo "  build            - Build the application"
+    @echo "  build-platform <platform>  - Build the application for a specified platform, linux/amd64 linux/arm64 and darwin/arm64"
+    @echo "  clean            - Clean build artifacts"
+    @echo "  release          - Builds all artifacts for release"
+    @echo "  docker-build     - Build Docker image with version tags"
+    @echo "  docker-push      - Push Docker images to registry"
+
+    @echo "\nVersioning:"
+    @echo "  version          - Show current version"
+    @echo "  version-bump     - Bump version (patch by default)"
+    @echo "  version-set      - Set specific version (e.g., just version-set 1.2.3)"
+    @echo "  version-major    - Bump major version"
+    @echo "  version-minor    - Bump minor version"
+    @echo "  version-patch    - Bump patch version"
+
+# Run recipe
+run:
+    #!/usr/bin/env bash
+    set -euo pipefail
+    echo "Running {{APP_NAME}} v{{VERSION}} locally..."
+    just build
+    ./{{BUILD_DIR}}/{{APP_NAME}} {{RUN_ARGS}}
+
+# Build recipe
+build:
+    #!/usr/bin/env bash
+    set -euo pipefail
```

could you explain what you do with that command? I see it everywhere

---

set -e: Exit immediately if any command returns a non-zero status (fail fast on errors).
set -u: Treat unset variables as an error and exit (catch typos and missing variables).
set -o pipefail: If any command in a pipeline fails, the whole pipeline fails (don't ignore errors in piped commands).